### ISSUE TITLE
Connection problems to Pi-based OpenVPN server

### ISSUE THREAD

**Issue by Darren-Hill** [lrusak/lrusak-openelec-addons#5]
Hi,

I have a Pi1 running an OpenVPN server set up using this tutorial - http://readwrite.com/2014/04/10/raspberry-pi-vpn-tutorial-server-secure-web-browsing

I'm trying to connect to it from a second Pi (a Pi Zero) running OpenElec 6.0.0, but am having issues. Using either a .ovpn file generated by the script in the tutorial, or a home-brewed one using the certificates and keys generated I get two errors at the end of the connection sequence:

ip: RTNETLINK answers: Invalid argument
ip: RTNETLINK answers: File exists

This was tested using ssh and the command line, as if I try running directly from the add-on settings page I just get "establishing connection" and then nothing else (and the IP stays as the original, not the DDNS of my server.

Attached is a transcript of a connection (with sensitive info removed, at least i hope) plus a copy of the .ovpn file (ditto, but renamed as a .txt to allow upload).

[vpn.txt](https://github.com/lrusak/lrusak-openelec-addons/files/94989/vpn.txt)
[client4.ovpn.txt](https://github.com/lrusak/lrusak-openelec-addons/files/95060/client4.ovpn.txt)

I can connect to this server from both Windows and Android devices using ovpn keys generated in the same way (including the attached one if tried separately).

Could you advise on what I may need to change to get it to work, or is this a question better asked at the OpenVPN forums?

Thanks

Darren

**Comment by lrusak** [comment]
Nothing jumps out at me

This is the client config I use (I use inline keys, but this shouldn't be an issue). I also use ```user``` and ```group``` but I can't see that being the issue. My vpn server supports lz4 compression. Maybe try adding the ```redirect-gateway def1``` option.

```
client
dev tun
proto udp
remote vps.xxxxxx.com 1194
resolv-retry infinite
nobind
user nobody
group nogroup
persist-key
persist-tun
;ca [inline]
;cert [inline]
;key [inline]
compress lz4
verb 3
redirect-gateway def1
key-direction 1

<ca>
ca goes here
</ca>

<cert>
cert goes here
</cert>

<key>
key goes here
</key>

<tls-auth>
tls-auth key goes here
</tls-auth>
```

**Comment by Darren-Hill** [comment]
Thanks for the feedback.

Unfortunately adding that option (plus the user and group ones) doesn't make a difference, same error results. The original script that is part of the server install generates .ovpn files with in-line keys, but that also suffers from the same error. I rewrote it using the sample config and separate files to keep things simpler, but I don't think that's the root of the issue here unfortunately.

I'll make a thread on the OpenVPN forums and see if anyone there can help me. It's just annoying, as it's so close but falling at the final fence for this particular device/client set-up, when it's worked fine for everything else I've tried to connect with it.

If I get a solution I'll update and let you know, just for reference in case anyone else ends up with the same issue.

**Comment by Darren-Hill** [comment]
I'm delighted to be able to report that I've resolved the issue.

The problem was the server, which was based on Raspbian Wheezy, and I think perhaps due to that and/or how the ovpn files were set up the dialogue with the client wasn't working. Anyway I've rebuilt the server now with Jessie-lite and using the set-up procedure below and with the new ovpn file that it generated the connection seems to be working fine for Windows, Android and most importantly Raspbian.

So my Pi Zero is now a happy bunny ready for my next business trip overseas on Monday, as am I that I can now watch iPlayer in France on a decent sized TV :) Thanks for a great add-on, my travelling media player is now complete!

https://www.digitalocean.com/community/tutorials/how-to-set-up-an-openvpn-server-on-debian-8
